phylonco-lphy/src/test/java/phylonco/lphy/evolution/copynumbermodel/ReadCopyProfileTest.java

```diff
@@ -0,0 +1,178 @@
+package phylonco.lphy.evolution.copynumbermodel;
+
+import lphy.base.evolution.Taxa;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class ReadCopyProfileTest {
+
+    private ReadCopyProfile reader;
+
+    @BeforeEach
+    void setUp() {
+        reader = new ReadCopyProfile(new lphy.core.model.Value<>("dummy", "dummy.txt"));
+    }
+
+    @Test
+    void testParseGinkgoFormat_WithRealData() {
+        String realFilePath = "/Users/ming/Desktop/Aug22/CRC09-GINKGO_24cells.CN.txt";
+
+        try {
+            List<String> sampleLines = readFirstLines(realFilePath, 31);
+            if (sampleLines.size() < 2) {
+                System.out.println("Real data file too small or not found, skipping test");
+                return;
+            }
+
+            List<String> cellNames = extractGinkgoCellNames(sampleLines.get(0));
+            if (cellNames.isEmpty()) {
+                System.out.println("File doesn't appear to be GINKGO format, skipping test");
+                return;
+            }
+
+            Taxa taxa = Taxa.createTaxa(cellNames.toArray(new String[0]));
+
+            System.out.println("=== Testing parseGinkgoFormat with Real Data ===");
+            System.out.println("Total lines used: " + sampleLines.size() + " (header + " + (sampleLines.size() - 1) + " data rows)");
+
+            // Test the parsing method
+            IntegerCharacterMatrix result = reader.parseGinkgoFormat(sampleLines, taxa, cellNames);
+
+            // Basic assertions
+            assertEquals(cellNames.size(), result.getTaxa().ntaxa(), "Cell count should match");
+            assertEquals(sampleLines.size() - 1, result.nchar().intValue(), "Bin count should match data lines");
+
+        } catch (Exception e) {
+            System.err.println("Real GINKGO data parsing test failed: " + e.getMessage());
+            // Don't fail test if file issues
+        }
+    }
+
+    @Test
+    void testParseSconce2Format_WithRealData() {
+        String realFilePath = "/Users/ming/Desktop/Aug22/CRC09-SCONCE2.CN.txt";
+
+        try {
+            List<String> cellNames = extractSconce2CellNames(realFilePath);
+            if (cellNames.isEmpty()) {
+                System.out.println("Real SCONCE2 data file not found, skipping test");
```

Comment:
I'd suggest logging the warning message "SCONCE2 data file not found, skipping test".